File: py/removeOuterParentheses.py

def removeOuterParentheses(S):
    
    l = []
    count = 0
    start = 0

    for i in range(len(S)):
        if S[i] == "(":
            count += 1
        elif S[i] == ")":
            count -= 1
        if count == 0:
            l.append(S[start+1:i])
            start = i + 1
    return "".join(l)


print(removeOuterParentheses("(()())(())(()(()))"))

File: js/removeOuterParentheses.js


function removeOuterParentheses(str) {

    l = [];
    count = 0;
    start = 0;

    for (let i = 0; i < str.length; ++i) {
        if (str[i] == "(") {
            count++;
        } else if (str[i] == ")") {
            count--;
        }
        if (count === 0) {
            l.push(str.substring(start+1, i));
            start = i + 1;
        }
    }
    return l.join('');

}

console.log(removeOuterParentheses("(()())(())"));
console.log('hi');